src/test/java/de/qabel/desktop/daemon/sync/worker/DefaultSyncerTest.java

```diff
@@ -128,16 +140,34 @@ public void stopStopsTransactions() throws Exception {
         syncer = new DefaultSyncer(config, new BoxVolumeStub(), manager);
         syncer.run();
 
-        waitUntil(() -> manager.getTransactions().size() == 1);
+        waitUntil(() -> manager.getTransactions().size() == 1, () -> manager.getTransactions().toString());
         syncer.stop();
         manager.cleanup();
         assertEquals(0, manager.getTransactions().size());
     }
 
     @Test
+    public void stopStopsRemoteWacher() throws Exception {
+        BoxNavigationStub nav = new BoxNavigationStub(null, BoxFileSystem.getRoot());
+        BoxVolumeStub volume = new BoxVolumeStub();
+        volume.rootNavigation = nav;
+        syncer = new DefaultSyncer(config, volume, manager);
+        syncer.run();
+
+        waitUntil(() -> manager.getTransactions().size() == 1 && syncer.isPolling());
+        syncer.stop();
+        manager.getHistory().clear();
+
+        nav.pushNotification(new BoxFile("a", "b", "c", 1L, 2L, new byte[0]), CREATE);
+        // wait for something to not happen?
+        Thread.sleep(100);
```

Comment:
At that point it is always empty. the `pushNotification` method calls a `notifyAsync` (to allow for navigation refreshes to happen even if the observers are slow) and that is asynchronously scheduled to the TransferManager ... and then added to the history. so a waitUntil(manage.getHisotry()::isEmpty) would be true without the fix